test/org/sagebionetworks/bridge/dynamodb/DynamoUserConsentDaoTest.java

```diff
@@ -90,11 +116,90 @@ public void canCountStudyParticipants() throws Exception {
         count = userConsentDao.getNumberOfParticipants(STUDY_IDENTIFIER);
         assertEquals("Correct number of participants", 5, count);
     }
+    
+    @Test
+    public void canGetAllConsentsInHistory() throws Exception {
+        final DynamoStudyConsent1 consent = createStudyConsent();
+        final StudyIdentifier studyId = new StudyIdentifierImpl(consent.getStudyKey());
+        for (int i=0; i < 5; i++) {
+            userConsentDao.giveConsent(HEALTH_CODE, consent, DateTime.now().getMillis());
+            userConsentDao.withdrawConsent(HEALTH_CODE, studyId, DateTime.now().getMillis());
+        }
+        userConsentDao.giveConsent(HEALTH_CODE, consent, DateTime.now().getMillis());
+        
+        List<UserConsent> consents = userConsentDao.getUserConsentHistory(HEALTH_CODE, studyId);
+        assertEquals(6, consents.size());
+        for (int i=0; i < 5; i++) {
+            assertNotNull(consents.get(i).getWithdrewOn());
+        }
+        assertNull(consents.get(consents.size()-1).getWithdrewOn());
+    }
 
-    private DynamoStudyConsent1 createStudyConsent() {
+    @Test
+    public void singleUserCanSignMultipleConsents() {
+        // Interestingly, you could sign the same study consent more than once.
+        DynamoStudyConsent1 consent = createStudyConsent(DateTime.now().getMillis());
+        StudyIdentifier studyId = new StudyIdentifierImpl(consent.getStudyKey());
+
+        long signedOn = DateTime.now().getMillis();
+        userConsentDao.giveConsent(HEALTH_CODE, consent, signedOn);
+        
+        // But you cannot sign the same study a second time
+        try {
+            userConsentDao.giveConsent(HEALTH_CODE, consent, signedOn);
+        } catch(BridgeServiceException e) {
+            assertEquals(409, e.getStatusCode());
+            assertEquals("Consent already exists.", e.getMessage());
+        }
+        
+        // Now create a different study consent, you should be able to consent again.
+        long secondConsentCreatedOn = DateTime.now().getMillis();
+        consent = createStudyConsent(secondConsentCreatedOn);
+        long signedOn2 = signedOn + (100000);
+        userConsentDao.giveConsent(HEALTH_CODE, consent, signedOn2);
+        
+        List<UserConsent> consents = userConsentDao.getUserConsentHistory(HEALTH_CODE, studyId);
+        assertEquals(2, consents.size());
+        assertEquals(signedOn, consents.get(0).getSignedOn());
+        assertEquals(signedOn2, consents.get(1).getSignedOn());
+        
+        // The active consent is the second signed consent.
+        UserConsent activeConsent = userConsentDao.getActiveUserConsent(HEALTH_CODE, studyId);
+        assertEquals(signedOn2, activeConsent.getSignedOn());
+        assertEquals(secondConsentCreatedOn, activeConsent.getConsentCreatedOn());
+    }
```

Comment:
My main concern for this test isn't the signing part, but the getting part, specifically that the query on signedOn for getUserConsent() works as intended. Can you add that to this test?